timelock-server/build.gradle

```diff
@@ -43,11 +45,32 @@ dependencies {
     testCompile group: 'org.mockito', name: 'mockito-core'
     testCompile project(":atlasdb-tests-shared")
 
-    integTestCompile group: 'io.dropwizard', name: 'dropwizard-testing'
+    integTestCompile group: 'io.dropwizard', name: 'dropwizard-testing', version: libVersions.timelock_dropwizard
 }
 
 distribution {
     serviceName "timelock-server"
     mainClass 'com.palantir.atlasdb.timelock.TimeLockServerLauncher'
     args 'server', 'var/conf/timelock.yml'
+    defaultJvmOpts "-javaagent:service/javaAgents/jetty-alpn-agent.jar"
```

Comment:
We definitely want to set min & max to the same value so that we avoid having to expand the heap as that requires an expensive `Full GC` (see several ~100ms pauses `Full GC (Ergonomics)` below from a recent test run).

```
2017-02-20T23:25:18.043+0000: 11902.330: [Full GC (Ergonomics) [PSYoungGen: 1345K->0K(29696K)] [ParOldGen: 86478K->20319K(112128K)] 87824K->20319K(141824K), [Metaspace: 41836K->41657K(1087488K)], 0.1119925 secs] [Times: user=0.23 sys=0.05, real=0.11 secs]
2017-02-21T10:14:49.331+0000: 50873.617: [Full GC (Ergonomics) [PSYoungGen: 512K->0K(125440K)] [ParOldGen: 112914K->23510K(114688K)] 113426K->23510K(240128K), [Metaspace: 41842K->41842K(1087488K)], 0.0827761 secs] [Times: user=0.18 sys=0.01, real=0.08 secs]
2017-02-21T14:35:47.501+0000: 66531.787: [Full GC (Ergonomics) [PSYoungGen: 1008K->0K(60928K)] [ParOldGen: 114963K->22286K(110592K)] 115972K->22286K(171520K), [Metaspace: 41873K->41873K(1087488K)], 0.0973755 secs] [Times: user=0.14 sys=0.00, real=0.09 secs]
2017-02-21T17:38:56.790+0000: 77521.076: [Full GC (Ergonomics) [PSYoungGen: 1104K->0K(37376K)] [ParOldGen: 110465K->23557K(108544K)] 111569K->23557K(145920K), [Metaspace: 41878K->41878K(1087488K)], 0.0721954 secs] [Times: user=0.14 sys=0.00, real=0.07 secs]
```

GC summary:
```
what           count          total          mean           max     stddev
gen0t           4120         16.801       0.00408         0.023     0.0014
gen1t              6          0.510       0.08496         0.112     0.0268
GCPause         4126         17.311       0.00420         0.112     0.0035
alloc           4120     213947.143      51.92892       170.500    34.4422
promo           4120        343.944       0.08348         7.146     0.1790
used0           4120     219741.247      53.33525       175.068    34.9468
used1              6        434.959      72.49316       112.269    49.4586
used            4126     479652.996     116.25133       266.163    45.0535
commit0         4120     222042.500      53.89381       181.500    35.2669
commit1         4120     430553.500     104.50328       112.500    10.4225
commit          4120     652596.000     158.39709       273.000    39.3772
used0AfterGC    4120       5794.839       1.40651         9.145     0.8145
used1AfterGC       6        114.729      19.12142        23.005     6.2507
usedAfterGC     4126     265714.431      64.40001       113.254    24.7004

alloc/elapsed_time    = 213947.143 MB /  77703.668 s  =   2.753 MB/s
alloc/tot_cpu_time    = 213947.143 MB / 310814.672 s  =   0.688 MB/s
alloc/mut_cpu_time    = 213947.143 MB / 310745.428 s  =   0.688 MB/s
promo/elapsed_time    =    343.944 MB /  77703.668 s  =   0.004 MB/s
promo/gc0_time        =    343.944 MB /     16.801 s  =  20.471 MB/s
gc_seq_load           =     69.244 s  / 310814.672 s  =   0.022%
gc_conc_load          =      0.000 s  / 310814.672 s  =   0.000%
gc_tot_load           =     69.244 s  / 310814.672 s  =   0.022%
```